src/notebooksyncagent.cpp

```diff
@@ -1287,9 +1286,13 @@ bool NotebookSyncAgent::deleteIncidences(const KCalendarCore::Incidence::List de
 {
     NOTEBOOK_FUNCTION_CALL_TRACE;
     bool success = true;
-    for (KCalendarCore::Incidence::Ptr doomed : deletedIncidences) {
-        mStorage->load(doomed->uid(), doomed->recurrenceId());
-        if (!mCalendar->deleteIncidence(mCalendar->incidence(doomed->uid(), doomed->recurrenceId()))) {
+    for (KCalendarCore::Incidence::Ptr incidence : deletedIncidences) {
+        KCalendarCore::Incidence::Ptr doomed = mCalendar->incidence(incidence->uid(), incidence->recurrenceId());
+        if (!doomed) {
+            mStorage->load(incidence->uid());
+            doomed = mCalendar->incidence(incidence->uid(), incidence->recurrenceId());
+        }
+        if (doomed && !mCalendar->deleteIncidence(doomed)) {
```

Comment:
Looks like this is a bit more than just adjusting the load() call, but guess it's just some extra protection.

Reply:
Yeh, I decided to rewrite a bit the routine. Technically it was correct as done before. But moving to load(uid), may call the load method for every exception which is useless. That's why I added a lookup first in the calendar and a load() only if necessary.